strategies/categorized/src/main/java/org/arquillian/smart/testing/strategies/categorized/CategorizedTestSelection.java

```diff
@@ -0,0 +1,57 @@
+package org.arquillian.smart.testing.strategies.categorized;
+
+import org.arquillian.smart.testing.TestSelection;
+
+import java.util.*;
+
+public class CategorizedTestSelection {
+
+    private final String className;
+    private boolean hasInheritedAllTestMethods;
+    private final List<String> appliedStrategies;
+    private final List<String> includedTestMethodNames;
+    private final List<String> excludedTestMethodNames;
+
+    public CategorizedTestSelection(String className, String... appliedStrategies) {
+        this.className = className;
+        hasInheritedAllTestMethods = false;
+        this.appliedStrategies = Arrays.asList(appliedStrategies);
+        includedTestMethodNames = new ArrayList<>();
+        excludedTestMethodNames = new ArrayList<>();
+    }
+
+    public void inheritAllTestMethods() {
+        this.hasInheritedAllTestMethods = true;
+    }
+
+    public boolean hasInheritedTestMethods() {
+        return hasInheritedAllTestMethods;
+
+    }
+
+    public void addIncludedTestMethodNames(Collection<String> testMethodNames) {
+        includedTestMethodNames.addAll(testMethodNames);
+    }
+
+    public void addExcludedTestMethodNames(Collection<String> excludedTestMethodNames) {
+        this.excludedTestMethodNames.addAll(excludedTestMethodNames);
+    }
+
+    public TestSelection toTestSelection() {
+        String[] appliedStrategies = this.appliedStrategies.stream().toArray(String[]::new);
+        if (hasInheritedAllTestMethods && excludedTestMethodNames.isEmpty()) {
+            return new TestSelection(className, appliedStrategies);
+        }
+        List<String> filteredTestMethodNames = new ArrayList<>(includedTestMethodNames);
+        filteredTestMethodNames.removeAll(excludedTestMethodNames);
```

Comment:
if you exclude all test methods, then it should return `TestSelection.NOT_MATCHED`, but you return `TestSelection` with an empty list of the test methods which means that it should execute the whole test class.

Reply:
Thanks for the feedback! Removed whole `CategorizedTestSelection` class in d1806ed